src/modules/map/components/openlayers/OpenLayersWMTSLayer.vue

```diff
@@ -44,43 +45,86 @@ const url = computed(() => {
 const layer = new TileLayer({
     id: layerId.value,
     opacity: opacity.value,
-    source: createXYZSourceForProjection(),
+    source: createWMTSSourceForProjection(),
 })
 
 // grabbing the map from the main OpenLayersMap component and use the composable that adds this layer to the map
 const olMap = inject('olMap', null)
 useAddLayerToMap(layer, olMap, zIndex)
 
 // reacting to changes accordingly
-watch(url, (newUrl) => layer.getSource().setUrl(newUrl))
+watch(url, (newUrl) => {
+    layer.getSource().setUrl(getWMTSUrl(newUrl))
+})
 watch(opacity, (newOpacity) => layer.setOpacity(newOpacity))
-watch(projection, () => layer.setSource(createXYZSourceForProjection()))
+watch(projection, () => layer.setSource(createWMTSSourceForProjection()))
+
+function getWMTSUrl(xyzUrl) {
+    return xyzUrl
+        .replace('{z}', '{TileMatrix}')
+        .replace('{x}', '{TileCol}')
+        .replace('{y}', '{TileRow}')
+}
 
 /**
- * Returns an OpenLayers XYZ source, with some customization depending on the projection being used.
+ * Returns an OpenLayers WMTS source, with some customization depending on the projection being
+ * used.
  *
  * If the projection is a CustomCoordinateSystem, it will set the extent of this projection to a
  * dedicated TileGrid object, meaning that tiles outside the extent won't be requested.
  *
  * If the projection is not a CustomCoordinateSystem, it will default to a worldwide coverage,
  * meaning no limit where tiles shouldn't be requested.
  *
- * @returns {XYZ}
+ * @returns {WMTSSource}
  */
-function createXYZSourceForProjection() {
-    let tileGrid = null
+function createWMTSSourceForProjection() {
+    let resolutions = []
+    let origin = null
+    let extent = null
+
     if (projection.value instanceof CustomCoordinateSystem) {
-        tileGrid = new TileGrid({
-            resolutions: projection.value.getResolutions(),
-            extent: projection.value.bounds.flatten,
-            origin: projection.value.getTileOrigin(),
-        })
+        resolutions = projection.value.getResolutions()
+        origin = projection.value.getTileOrigin()
+        extent = projection.value.bounds.flatten
+    } else {
+        // TODO(IS): move this to the projection class?
+        extent = projection.value.bounds.flatten
+        origin = getTopLeft(extent)
+        const size = getWidth(extent) / 256
+        resolutions = []
+        for (let z = 0; z < 19; ++z) {
+            resolutions[z] = size / Math.pow(2, z)
+        }
+    }
```

Comment:
As you stated in your TODO, I think this makes sense to move this resolution definition, and zoom level array, into the CoordinateSystem class hierarchy.